src/Tmds.Ssh/UserAuthentication.cs

```diff
@@ -29,12 +29,28 @@ private async static Task PerformDefaultAuthentication(SshConnection connection,
             ParseServiceAccept(serviceAcceptMsg);
         }
 
-        UserAuthContext context = new UserAuthContext(connection, userName, logger);
+        UserAuthContext context = new UserAuthContext(connection, userName, publicKeyAcceptedAlgorithms, minimumRSAKeySize, logger);
+
+        bool authSuccess = false;
+
+        // gssapi-with-mic may require interaction with the ticket server.
```

Comment:
Does this mean we should change the default back to not using `gssapi-with-mic` like how ssh works? This would save a network roundtrip unless the user actually opts into it.

Reply:
After looking more closely at the `gssapi-with-mic` auth, I removed this because I don't think we'll do any interaction with the ticket server until the server sent a response that tells us it accepts `gssapi-with-mic`.

Reply:
Ah that's a good point, I forgot the first message is the OID negotiation so yea `gssapi-with-mic` won't do the TGT exchange until that succeeds.